### PR TITLE
PR 60

### PR DESCRIPTION
**Pull request by ericsorenson**
### Stories & Bugs

- [7555 BUG: Pretrial Memos are accessible by public](https://github.com/flexion/ef-cms/issues/7555) 
- [7549 BUG: Service stamp turning vertical](https://github.com/flexion/ef-cms/issues/7549) 
- [7534 BUG: Today's Orders not Left Aligned](https://github.com/flexion/ef-cms/issues/7534) 
- [7533 BUG: Error received when serving document with extremely long title ](https://github.com/flexion/ef-cms/issues/7533) 
- [7524 BUG: Edit Docket Entry - Page never loads; receiving spinner and error ](https://github.com/flexion/ef-cms/issues/7524) 
- [7509 BUG: Validation error on messages](https://github.com/flexion/ef-cms/issues/7509) 
- [7508 BUG: Edit Docket Entry: Required fields not displaying when doc type is updated to Scenario H document](https://github.com/flexion/ef-cms/issues/7508) 
- [7506 BUG: Edit Docket Entry: Not retaining all data in fields for Motion for Leave to File and causing errors when changing doc type](https://github.com/flexion/ef-cms/issues/7506) 
- [7473 BUG: Edit Docket Entry: Unable to select previous document when doc type is updated to Reply ](https://github.com/flexion/ef-cms/issues/7473) 
- [7472 BUG: Edit Docket Entry: When editing doc type to a court-issued code that requires cover sheet, updated PDF does not include cover sheet](https://github.com/flexion/ef-cms/issues/7472) 
- [7452 BUG: Completed messages still showing in Inbox](https://github.com/flexion/ef-cms/issues/7452) 
- [7304 BUG: MISCL code should not be displayed in document type dropdown for paper-filings ](https://github.com/flexion/ef-cms/issues/7304) 
- [7287 Practitioner w/o Email Address: Default to Paper Service](https://github.com/flexion/ef-cms/issues/7287) 

### Refactoring

- Run all of the cypress smoke tests during the build process as e2e tests
- Generic action for computing dates from forms
- Record all of our circle environment variables and their values
- Rename `saveWorkItemForNonPaper` to `saveWorkItemForPaper` :)
- Automate reseeding local database
- Update "docket number already exists" check to increment the docket number until a valid one is found instead of throwing an error
- Quick Perf Win: Refactor aggregateCaseItems

### Data Migrations

- 0014-practitioner-service-indicator.js
- 0015-practitioner-case-service-indicator.js
- 0016-hearings-proceeding-type.js

### CODE REVIEW

**Comment by klohman** [comment]
## PR 60 Release Notes
### Internal

**Improvement**: Practitioners without an email address on record will default to paper service
* Migrated practitioners without an email address will have paper service for all of their associated cases

**Bug Fix**: Character limits on fields were not throwing validation errors
* Message input fields can now accept up to 700 characters, and will show a helpful error message when the limit is reached.
* Document free text fields can now accept up to 1000 characters, and will show a helpful error message when the limit is reached.
* The complete document title now accepts up to 3000 characters total, and will inform the user when the limit is reached.

**Bug Fix**: Several fixes were made to the Edit Docket Entry feature
* All data entered when adding a docket entry is now retained on the Edit Docket Entry screen
* All required fields are now displayed when updating a document type on the Edit Docket Entry screen
* A cover sheet is now being applied when a docket entry is updated to event codes HE, TE, USCA, RM, ES
* Some migrated docket entries were causing an error when accessing the Edit Docket Entry screen. The undefined values have been identified and will no longer produce an error.

**Bug Fix**: Completed messages were still showing in user Inboxes
* When a message is completed, it is now removed from all users’ inboxes, regardless of who completed it.

**Bug Fix**: Service stamps were displaying vertically on rotated PDFs
* Service stamps and signatures that were displaying vertically when applied to PDFs will now display horizontally only.

**Bug Fix**: Pretrial Memorandums are no longer accessible by the public
* Some Pretrial Memorandum docket entries were hyperlinked for public users, and the PDF document was visible. Public users and those not associated with a case no longer can view a Pretrial Memorandum.

**Bug Fix**: Alignment corrected on Today’s Orders
* Order titles that expand to multiple lines on the Today’s Orders table are now left-aligned.  

**Bug Fix**: MISCL event code is no longer an option when adding a paper-filing
* The MISCL code has been removed from the document type dropdown. The code will be automatically applied when MISC is selected as the document type and the filing status is “Lodged”


### External / public

**Bug Fix**: Pretrial Memorandums PDFs were accessible to the public
* Some Pretrial Memorandum docket entries were hyperlinked for public users, and the PDF document was accessible. This document type is now not accessible to the public.

**Bug Fix**: Order title on Today’s Orders table was not left-aligned
* Order titles that expand to multiple lines on the Today’s Orders table are now left-aligned.  

**Bug Fix**: Character limits on fields were not throwing validation errors 
* Document free text fields can now accept up to 1000 characters, and will show a helpful error message when the limit is reached.
* The complete document title now accepts up to 3000 characters total, and will inform the user when the limit is reached.

**Comment by julialeague** [comment]
@ericsorenson Are there any changes in this PR that require additional/manual deployment steps?

**Comment by ericsorenson** [comment]
> @ericsorenson Are there any changes in this PR that require additional/manual deployment steps?

There are not.

File: iam/terraform/account-specific/main/circle-ci.tf [hunk 1]
```diff
@@ -70,22 +70,22 @@ resource "aws_iam_policy" "circle_ci_policy" {
...
-        "cognito-idp:SignUp",
         "cognito-idp:ListUserPools",
         "cognito-idp:ListUserPoolClients",
         "cognito-idp:CreateUserPoolClient",
         "cognito-idp:DescribeUserPoolDomain",
         "cognito-idp:SetUICustomization",
         "cognito-idp:DeleteUserPoolDomain",
         "cognito-idp:GetUserPoolMfaConfig",
-        "cognito-idp:SetUserPoolMfaConfig"
+        "cognito-idp:SetUserPoolMfaConfig",
```
Comment (julialeague) [inline iam/terraform/account-specific/main/circle-ci.tf:86]: Don't we need to separately run the `account-specific` deploy to apply these policy updates, or is this somehow covered in the CircleCI deployment?

Comment (ericsorenson) [reply iam/terraform/account-specific/main/circle-ci.tf:86]: These changes won't apply to existing environments, only new environments.

Comment (mmarcotte) [reply iam/terraform/account-specific/main/circle-ci.tf:86]: That's a good point Julia. It's something we ran into when rebuilding Dev. It's a step we ought to have to perform for all environments Thank you for flagging.

**Comment by julialeague** [comment]
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] If changes were made in the UI (if not, skip):
  - [x] If desired, run `docker system prune` to remove any unused docker images from previous code review.
  - [x] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
  - [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
	- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
	- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
	- [x] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
	- [x] user-visible changes (including API users like the IRS) are documented in CHANGELOG.md (which is linked from US Tax Court website).
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Review by julialeague — APPROVED** [review]
@mmarcotte, I recommend you accept these changes into the court's environments. Thank you!

**Pull request merged**